product-mini/platforms/esp-idf/CMakeLists.txt

```diff
@@ -3,53 +3,57 @@
 
 # from ESP-IDF 4.0 examples/build_system/cmake/idf_as_lib
 cmake_minimum_required(VERSION 3.5)
+project(wamr_on_esp32c3)
 
-project(wamr_esp_idf C)
+enable_language(ASM)
 
-enable_language (ASM)
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3")
+
+if("${IDF_TARGET}" STREQUAL "")
+    message(FATAL_ERROR "You need to set IDF_TARGET to your target string")
+endif()
 
 # Include for ESP-IDF build system functions
 include($ENV{IDF_PATH}/tools/cmake/idf.cmake)
-
-# Create idf::esp32 and idf::freertos static libraries
-idf_build_process(esp32
-                  # try and trim the build; additional components
-                  # will be included as needed based on dependency tree
-                  #
-                  # although esptool_py does not generate static library,
-                  # processing the component is needed for flashing related
-                  # targets and file generation
-                  COMPONENTS esp32 freertos esptool_py
-                  SDKCONFIG ${CMAKE_BINARY_DIR}/sdkconfig
-                  BUILD_DIR ${CMAKE_BINARY_DIR})
-
-include_directories(build/config
-                    xtensa/include
-                    $ENV{IDF_PATH}/components/esp32/include
-                    $ENV{IDF_PATH}/components/esp_common/include
-                    $ENV{IDF_PATH}/components/esp_rom/include
-                    $ENV{IDF_PATH}/components/freertos/include
-                    $ENV{IDF_PATH}/components/heap/include
-                    $ENV{IDF_PATH}/components/soc/esp32/include
-                    $ENV{IDF_PATH}/components/xtensa/include
-                    $ENV{IDF_PATH}/components/xtensa/esp32/include)
+# Create idf::esp32c3 and idf::freertos static libraries
+idf_build_process(${IDF_TARGET}
+                # try and trim the build; additional components
+                # will be included as needed based on dependency tree
+                #
+                # although esptool_py does not generate static library,
+                # processing the component is needed for flashing related
+                # targets and file generation
+                COMPONENTS ${IDF_TARGET} freertos esptool_py
+                SDKCONFIG ${CMAKE_BINARY_DIR}/sdkconfig
+                BUILD_DIR ${CMAKE_BINARY_DIR})
+
+# Set WAMR's build options
+if("${IDF_TARGET}" STREQUAL "esp32c3")
+    set(WAMR_BUILD_TARGET "RISCV32")
+else()
+    set(WAMR_BUILD_TARGET "XTENSA")
+    add_compile_options(-DWAMR_BUILD_TARGET_XTENSA=1)
+endif()
 
 set(WAMR_BUILD_PLATFORM "esp-idf")
-set(WAMR_BUILD_TARGET "XTENSA")
 set(WAMR_BUILD_INTERP 1)
-set(WAMR_BUILD_FAST_INTERP 1)
-set(WAMR_BUILD_AOT 1)
+set(WAMR_BUILD_FAST_INTERP 0)
+set(WAMR_BUILD_AOT 0)
 set(WAMR_BUILD_LIBC_BUILTIN 1)
-set(WAMR_BUILD_LIBC_WASI 0)
+set(WAMR_BUILD_APP_FRAMEWORK 0)
```

Comment:
Had better use
```cmake
if (NOT DEFINED WAMR_BUILD_XXX)
  set (WAMR_BUILD_XXX 1/0)
endif ()
```
instead, so developer can use `cmake -DWAMR_BUILD_XXX=1 or 0` to control the features.